### PR TITLE
server: make TestRegionSynce stable

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve?
ref #4348

### What is changed and how it works?
add more failpoint

### Check List

- Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Review by HunDunDM — APPROVED** [review]

File: server/region_syncer/server.go [hunk 1]
```diff
@@ -153,6 +153,18 @@ func (s *RegionSyncer) RunServer(ctx context.Context, regionNotifier <-chan *cor
 	}
 }
 
+// GetAllDownstreamNames tries to get the all bind stream's name.
+// Only for test
+func (s *RegionSyncer) GetAllDownstreamNames() []string {
+	var names []string
```
Comment (JmPotato) [inline server/region_syncer/server.go:159]: Better initialize it with a capability.

File: server/region_syncer/server.go [hunk 2]
```diff
@@ -153,6 +153,18 @@ func (s *RegionSyncer) RunServer(ctx context.Context, regionNotifier <-chan *cor
...
+	s.mu.RLock()
+	for name := range s.mu.streams {
+		names = append(names, name)
+	}
```
Comment (JmPotato) [inline server/region_syncer/server.go:163]: What about using `copy()`?

File: tests/cluster.go [hunk 3]
```diff
@@ -534,6 +534,31 @@ func (c *TestCluster) WaitLeader(ops ...WaitOption) string {
...
+	for i := 0; i < option.retryTimes; i++ {
+		name := c.GetLeader()
+		if len(name) == 0 {
+			time.Sleep(option.waitInterval)
+			continue
+		}
+		leaderServer := c.GetServer(name)
+		clus := leaderServer.GetServer().GetRaftCluster()
+		if clus != nil {
+			if len(clus.GetRegionSyncer().GetAllDownstreamNames()) == n {
```
Comment (JmPotato) [inline tests/cluster.go:555]: Since we only use its length, why bother copying the whole slice?

Comment (nolouch) [reply tests/cluster.go:555]: I think it is more clear to return the name of the current downstream than to return only the number, although it is only used in test now.

File: server/region_syncer/server.go (hunk 2 above)
Comment (nolouch) [reply server/region_syncer/server.go:163]: this is a map, do you mean copy map?

**Pull request merged**